docs/inputs.rst

```diff
@@ -0,0 +1,215 @@
+.. _inputs:
+
+Inputs
+======
+
+Melody inputs allow the user to specify the space that they would like
+to search. Inputs are specified as a list of individual input objects.
+
+Supported
++++++++++
+
+The following input classes are currently supported.
+
+Fixed
+*****
+
+.. autoclass:: melody.inputs.Fixed
+    :members:
+
+For example
+::
+
+   >>> from melody.inputs import Fixed
+   >>> inputs = [Fixed(name="option1", value="value1")]
+
+The above example will generate an input named ``option1`` with the value
+``value1``.
+
+Switch
+******
+
+.. autoclass:: melody.inputs.Switch
+    :members:
+
+For example
+::
+   
+   >>> from melody.inputs import Switch
+   >>> inputs = [Switch(name="option1", off="dark", on="light")]
+
+The above example will generate an input named ``option1`` with two
+values, ``dark`` and ``light``. If one of the arguments ``off`` or
+``on`` is not provided then an empty string is returned for that
+option. If both of the arguments ``off`` and ``on`` are not provided
+then a ``Runtime`` exception is raised.
+
+Choice
+******
+
+.. autoclass:: melody.inputs.Choice
+    :members:
+
+For example
+::
+   
+   >>> from melody.inputs import Choice
+   >>> inputs = [Choice(name="input2", inputs=["a", "b", "c"])]
+
+The above example will generate an input named ``input2`` with three
+values, ``a``, ``b`` and ``c``. The list can be arbitrarily long.
+
+The optional ``pre`` argument will prepend all inputs with the value
+contained in the string. For example
+::
+
+   >>> from melody.inputs import Choice
+   >>> inputs = [Choice(name="in", inputs=["a", "b", "c"], pre="val_")]
+
+The above example will generate an input named ``in`` with three
+values, ``val_a``, ``val_b`` and ``val_c``.
+
+IntRange
+********
+
+.. autoclass:: melody.inputs.IntRange
+    :members:
+
+For example
+::
+   
+   >>> from melody.inputs import IntRange
+   >>> inputs = [IntRange(name="range1", low=0, high=3, step=1)]
+
+The above example will generate an input named ``range1`` with three
+integer values, ``0``, ``1`` and ``2``.
+
+.. warning::
+
+   You might expect the integer 3 to appear, however consistency is
+   being maintained with the Python range function which will not
+   return this value.
+
+IntRange
+********
+
+.. autoclass:: melody.inputs.FloatRange
+    :members:
+
+For example
+::
+   
+   >>> from melody.inputs import FloatRange
+   >>> inputs = [FloatRange(name="range2", low=0.0, high=0.4, step=0.1)]
+
+
+The above example will generate an input named ``range2`` with four
+floating point values, ``0.0``, ``0.1``, ``0.2`` and ``0.3``.
+
+.. warning::
+
+   You might expect the value 0.4 to appear, however consistency is
```

Comment:
Suggest "however, in keeping with the Python range function, this value is excluded."